### DISCUSSION TITLE
Filtered group by, how?

### DISCUSSION THREAD

**Discussion by kde99** [ravendb/ravendb#19552] [Q&A] [answered]
Hello everyone,

I have a collection called `events`, which collects certain events that happened. I want to write a query that given a time period (from -> to) will give me the count a given event type had been logged. Naively it should look like this:
```sql
from "events"
group by  event_type_id
where event_time >= '2024-10-07T00:00:00' and event_time <= '2024-10-08T00:00:00'
order by event_type_id
select event_type_id, count(event_type_id)
```
but this query fails with: `Field 'event_time' is neither an aggregation operation nor part of the group by key`. Searching for this specific message hasn't brought me any closer to solving it, only finding unanswered questions, and hacky solutions where they had to specify fixed filters on indexes. 

I would want my filter to be dynamically changeable and not be static, without having to create multiple indexes or reaggregating client side.

There is a possible solution from 2 years ago https://github.com/ravendb/ravendb/discussions/14740, but isn't there any better ways yet?

**Comment by ayende** [comment, marked as answer]
The issue is that the way RavenDB handles grouping is at indexing time, so you need to account for that.

There are two ways to do that, you can use:

```
from "events"
where event_time between '2024-10-07T00:00:00' and  '2024-10-08T00:00:00'
select facet(event_type_id)
```

Which will give you the data you want. 
Or you can use map reduce index and group by day, then get the value on a per day basis.

**Reply by kde99** [reply]
Thank you for your straight on point answer!
